### ISSUE TITLE
`Use unless` doesn't fire for compound `else` expressions

### ISSUE THREAD

**Issue by LightAndLight** [ndmitchell/hlint#1671]
```
$ hlint --version
HLint v3.10, (C) Neil Mitchell 2006-2025
```

Example:

```
module Test where

test1 = if x then pure () else y

test2 = if x then pure () else y x
```

`test1`'s RHS generates `Warning: Use unless`, but `test2`'s RHS doesn't trigger the warning.
